test/integration_test.rb

```diff
@@ -261,14 +264,16 @@ def test_rebase_only_with_unstaged_changes
         RakeCommit::Shell.system "git add ."
         RakeCommit::Shell.system "git commit -m 'Added bar.baz (this commit should be preserved!)'"
         RakeCommit::Shell.system 'echo "change" >> bar.baz'
-        RakeCommit::Shell.system "git status"
-        RakeCommit::Shell.system "yes | ../../../bin/rake_commit --rebase-only"
-        files = RakeCommit::Shell.backtick("git ls-files")
-        assert files.include?("foo.bar")
+
+        log_lines_before = RakeCommit::Shell.backtick("git log --pretty=oneline").split("\n")
+        begin
+          output = RakeCommit::Shell.system "yes | ../../../bin/rake_commit --rebase-only"
+          fail
+        rescue
+        end
 
         log_lines = RakeCommit::Shell.backtick("git log --pretty=oneline").split("\n")
-        assert_equal 4, log_lines.size
-        assert_match /should be preserved/, log_lines[1]
+        assert_equal log_lines_before, log_lines
       end
     end
   end
```

Comment:
Can you update the tests to also check for the error message? Something like `assert_match /You have uncommitted changes/, output`.